File: desh/ege2026kp/16solve/16-69.py

m = {}
def F( n ):
  if n in m:
    return m[n]
  if n < 4:
    res = n - 1
  elif n % 3 == 0:
    res = n + 2*F(n-1)
  else:
    res = F(n-2) + F(n-3)
  m[n] = res
  return res

print( sum( map(int, str(F(25))) ) )

File: desh/ege2026kp/16solve/16-69.pas
function F( n: integer ): integer; 
begin 
   if n < 4 then  
     Result := n-1 
   else if n mod 3 = 0  then  
     Result := n + 2*F(n-1) 
   else 
     Result := F(n-2) + F(n-3) 
end; 
begin 
   writeln( F(25).ToString.Select(x->StrToInt(x)).Sum ); 
end. 